webapp/tests/karma/ts/modules/tasks/tasks.component.spec.ts

```diff
@@ -0,0 +1,227 @@
+import { ComponentFixture, fakeAsync, flush, TestBed, tick } from '@angular/core/testing';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { TranslateFakeLoader, TranslateLoader, TranslateModule } from '@ngx-translate/core';
+import { expect } from 'chai';
+import * as moment from 'moment';
+import sinon from 'sinon';
+
+import { ChangesService } from '@mm-services/changes.service';
+import { ContactTypesService } from '@mm-services/contact-types.service';
+import { RulesEngineService } from '@mm-services/rules-engine.service';
+import { TasksActions } from '@mm-actions/tasks';
+import { TelemetryService } from '@mm-services/telemetry.service';
+import { TourService } from '@mm-services/tour.service';
+import { TasksComponent } from '@mm-modules/tasks/tasks.component';
+import { NavigationComponent } from '@mm-components/navigation/navigation.component';
+import { Selectors } from '@mm-selectors/index';
+
+describe('TasksComponent', () => {
+  let getComponent;
+  let Changes;
+  let RulesEngine;
+  let Telemetry;
+  let Tour;
+  let contactTypesService;
+  let clock;
+  let store;
+
+  let component: TasksComponent;
+  let fixture: ComponentFixture<TasksComponent>;
+
+  beforeEach(async () => {
+    Changes = sinon.stub();
+    RulesEngine = {
+      isEnabled: sinon.stub().resolves(true),
+      fetchTaskDocsForAllContacts: sinon.stub().resolves([]),
+    };
+
+    Tour = sinon.stub();
+    Telemetry = { record: sinon.stub() };
+    contactTypesService = {
+      includes: sinon.stub(),
+    };
+
+    TestBed.configureTestingModule({
+      imports: [
+        TranslateModule.forRoot({ loader: { provide: TranslateLoader, useClass: TranslateFakeLoader } }),
+        RouterTestingModule,
+      ],
+      providers: [
+        provideMockStore(),
+        { provide: ChangesService, useValue: { subscribe: Changes }},
+        { provide: RulesEngineService, useValue: RulesEngine },
+        { provide: TelemetryService, useValue: Telemetry },
+        { provide: TourService, useValue: { startIfNeeded: TourService } },
```

Comment:
Shouldn't this one also use `Tour` stub? Since it's been used in line 101 🤔 I can't find where it's been injected in the test module/component.

Reply:
Yeap, changed!